api/institutions/authentication.py

```diff
@@ -334,14 +334,13 @@ def authenticate(self, request):
             user.save()
 
             # Send confirmation email for all three: created, confirmed and claimed
-            send_mail(
-                to_addr=user.username,
-                mail=WELCOME_OSF4I,
-                user=user,
-                domain=DOMAIN,
-                osf_support_email=OSF_SUPPORT_EMAIL,
-                storage_flag_is_active=flag_is_active(request, features.STORAGE_I18N),
-            )
+            notification_type = NotificationType.objects.filter(name='welcome_osf4i')
+            if not notification_type.exists():
```

Comment:
Since we should always have notification types loaded in production we don't have to check here. unhandled exception is actually better for this use case as it would truly mean the system hadn't been configure properly.